let DEBUG_MODE,ValidationError,CustomErrorHandler;_bf5‍.x([["default",()=>_bf5‍.o]]);_bf5‍.w("../config",[["DEBUG_MODE",["DEBUG_MODE"],function(v){DEBUG_MODE=v}]]);_bf5‍.w("joi",[["ValidationError",["ValidationError"],function(v){ValidationError=v}]]);_bf5‍.w("../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);



const errorHandler =(err, req, res, next)=>{
    let statuscode =500;
    let data ={
        message:'Internal Server Error',
        ...(DEBUG_MODE === 'true' && { originalError : err.message})
    }

    //validation error handling
    if(err instanceof ValidationError){
        statuscode=422;
        data ={
            message : err.message
        }
    }

    //user is already in database error handling
    if(err instanceof CustomErrorHandler){
        statuscode = err.status;
        data = {
            message : err.message
        }
    }



    return res.status(statuscode).json(data);
}

_bf5‍.d(errorHandler);